scipy/stats/_stats_py.py

```diff
@@ -9717,6 +9717,375 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
+       Form," The Annals of Mathematical Statistics, vol. 7, no. 3,
+       pp. 122-128, 1936, Accessed: Sep. 18, 2019. [Online]. Available:
+       https://www.jstor.org/stable/2957563.
+    .. [3] H. A. David and H. N. Nagaraja, "Order Statistics in Nonparametric
+       Inference" in Order Statistics, John Wiley & Sons, Ltd, 2005, pp.
+       159-170. Available:
+       https://onlinelibrary.wiley.com/doi/10.1002/0471722162.ch7.
+    .. [4] N. Hutson, A. Hutson, L. Yan, "QuantileNPCI: Nonparametric
+       Confidence Intervals for Quantiles," R package,
+       https://cran.r-project.org/package=QuantileNPCI
+    .. [5] M. Mayer, "confintr: Confidence Intervals," R package,
+       https://cran.r-project.org/package=confintr
+
+
+    Examples
+    --------
+
+    Suppose we wish to test the null hypothesis that the median of a population
+    is equal to 0.5. We choose a confidence level of 95%; that is, we will
+    reject the null hypothesis in favor of the alternative if the p-value is
+    less than 0.05.
+    
+    When testing random variates from the standard uniform distribution, which
+    has a median of 0.5, we expect the data to be consistent with the null
+    hypothesis most of the time.
+    
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> rng = np.random.default_rng()
+    >>> rvs = stats.uniform.rvs(size=100, random_state=rng)
+    >>> stats.quantile_test(rvs, q=0.5, p=0.5)
+    QuantileTestResult(statistic1=46, statistic2=46, pvalue=0.18410080866334788)
+    
+    As expected, the p-value of 0.018 is not below our threshold of 0.05, so
+    we cannot reject the null hypothesis.
+    
+    When testing data from the standard *normal* distribution, which has a 
+    median of 0, we would expect the null hypothesis to be rejected.
+    
+    >>> rvs = stats.norm.rvs(size=100, random_state=rng)
+    >>> stats.quantile_test(rvs, q=0.5, p=0.5)
+    QuantileTestResult(statistic1=62, statistic2=62, pvalue=0.0060164878626817395)
+    
+    Indeed, the p-value is lower than our threshold of 0.01, so we reject the
+    null hypothesis in favor of the default "two-sided" alternative: the median
+    of the population is *not* equal to 0.5.
+
+    However, suppose we were to test the null hypothesis against the
+    one-sided alternative that the median of the population is *greater* than
+    0.5. Since the median of the standard normal is less than 0.5, we would not
+    expect the null hypothesis to be rejected.
+    
+    >>> stats.quantile_test(rvs, q=0.5, p=0.5, alternative='greater')
+    QuantileTestResult(statistic1=82, statistic2=82, pvalue=0.9895106321610742)
+    
+    Unsurprisingly, with a p-value greater than our threshold, we would not
+    reject the null hypothesis.
+    
+    The quantile test can be used for any quantile, not only the median. For
+    example, we can test whether the third quartile of uniform distribution is
+    larger than 0.6.
```

Comment:
Ack. I agree. Why do you say "This is not incorrect as written" though? Your suggested change seems correct to me...

Reply:
Both are correct. The distribution underlying your sample *is* (happens to be) the uniform distribution. But typically, the user will not know what their distribution is, and it will not usually be the uniform distribution. So the point is to emphasize what is relevant to the user.

Reply:
That makes sense to me

Reply:
OK, I implemented this.